### PR TITLE
[IMPROVE] Allow dragging of images and text from browsers

### LINKED ISSUE #14358: Image drag & drop from Firefox onto a message list is not handled
### Description:

Dragging an image from the browser into a chat in the same browser is not handled well.

### Steps to reproduce:

1. Open any image in Firefox (66.0.3 for me).
2. Open Rocket in a new tab in the same browser; open a chat room.
3. Drag the image from 1 into the tab from 2; aim for the message list.

### Expected behavior:

The image (or its URL) is sent (resp. inserted into the input box).

For reference: dropping the image on the input area will pase its URL into the input box. Not spectacular, but reasonable behaviour.

### Actual behavior:

Firefox navigates to the image URL, leaving Rocket.

### Server Setup Information:

- Version of Rocket.Chat Server: 1.0.2
- Operating System: ???
- Deployment Method: ???
- Number of Running Instances: ???
- DB Replicaset Oplog: ???
- NodeJS Version: ???
- MongoDB Version: ???

(I'm not an administrator of that server, so I can't determine this information.)

### PR DESCRIPTION
**Pull request by MartinSchoeler**
Closes #14358

### CODE REVIEW

File: app/ui/client/views/app/room.js [hunk 1]
```diff
@@ -224,6 +224,35 @@ function roomMaxAge(room) {
...
+	let response;
+	try {
+		response = await fetch(url);
+	} catch (error) {
+		if (error.toString().includes('TypeError')) {
+			return url;
+		}
+		throw error;
+	}
+	const fileExtension = url.split('.').pop().split(/\#|\?/)[0] || 'jpg';
```
Comment (MartinSchoeler) [inline app/ui/client/views/app/room.js]: Need to improve this chunk to better detect the file extention, maybe even get it from the allowed file types setting

File: app/ui/client/views/app/room.js [hunk 2]
```diff
@@ -548,14 +578,35 @@ export const dropzoneEvents = {
...
 
 		e.stopPropagation();
 
-		const files = (e.dataTransfer != null ? e.dataTransfer.files : undefined) || [];
+		let files = (e.dataTransfer != null ? e.dataTransfer.files : undefined) || [];
+
+		if (files.length < 1) {
+			const transferData = e.dataTransfer.getData('text') !== '' ? e.dataTransfer.getData('text') : e.dataTransfer.getData('url');
+			if (e.dataTransfer.types.includes('text/uri-list')) {
+				const dropContext = $('<div>').append(e.dataTransfer.getData('text/html'));
```
Comment (tassoevan) [inline app/ui/client/views/app/room.js]: We've been avoiding to use jQuery whenever is possible to. Here, the vanilla alternative would be a [DocumentFragment](https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment) for `dropContext` and queries using the `querySelector()` API. However, jQuery safely handle nulls -- which allows you to write long chains of methods --, therefore you need to break the calls into smaller steps.

File: app/ui/client/views/app/room.js [hunk 3]
```diff
@@ -224,6 +224,34 @@ function roomMaxAge(room) {
 	}
 }
 
+async function createFileFromUrl(url) {
+	let response;
+	try {
+		response = await fetch(url);
+	} catch (error) {
+		if (error.toString().includes('TypeError')) {
```
Comment (tassoevan) [inline app/ui/client/views/app/room.js]: When is this triggered? Can we grant `url` type in the outer scope?

Comment (MartinSchoeler) [reply app/ui/client/views/app/room.js]: @tassoevan https://github.com/RocketChat/Rocket.Chat/blob/fix-clipboard-drag/app/ui/client/views/app/room.js#L599 I Check this here

**Pull request merged**